core/src/test/java/com/linecorp/armeria/internal/server/annotation/EmptyParameterWithDefaultAnnotationTest.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.server.annotation;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.annotation.Default;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.server.annotation.Param;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+class EmptyParameterWithDefaultAnnotationTest {
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.annotatedService("/default", new EmptyParameterService());
+        }
+    };
+
+    static class EmptyParameterService {
+        @Get("/string")
+        public String string(@Default @Param String param) {
+            return String.valueOf(param);
+        }
+
+        @Get("/integer")
+        public String integer(@Default @Param Integer param) {
+            return String.valueOf(param);
+        }
+
+        @Get("/primitive")
+        public String primitive(@Default @Param int param) {
+            return String.valueOf(param);
+        }
+
+        @Get("/string-list")
+        public HttpResponse stringList(@Default @Param("param") List<String> params) {
+            return HttpResponse.ofJson(params);
+        }
+
+        @Get("/integer-list")
+        public HttpResponse integerList(@Default @Param("param") List<Integer> params) {
+            return HttpResponse.ofJson(params);
+        }
+
+        @Get("/integer-list-with-default")
+        public HttpResponse integerListWithDefault(@Default("1") @Param("param") List<Integer> params) {
+            return HttpResponse.ofJson(params);
+        }
+    }
+
+    private static final String NULL = "null";
+
+    private final WebClient client = WebClient.of(server.httpUri() + "/default");
+
+    @Test
+    void testEmptyStringParameterIsEmpty() {
+        AggregatedHttpResponse res = aggregate(client.get("/string?param=v"));
```

Comment:
Instead, we can use `server.blockingClient()`.

Reply:
@ikhoon
I failed to replace it into `server.blockingCleint()` because it seems that the `DefaultBlockingWebClient` has `/` as `absolutePathRef()`.

This fails with 404 Not Found:
```java
final BlockingWebClient blockingWebClient = server.blockingWebClient();
AggregatedHttpResponse res = blockingWebClient.get("/string", 
                                                   QueryParams.of("param", "v"));
```

Should I append all the prefixes in front of request url like:
```java
final BlockingWebClient blockingWebClient = server.blockingWebClient();
AggregatedHttpResponse res = blockingWebClient.get("/default/string", 
                                                   QueryParams.of("param", "v"));
```
?

Reply:
We don't have an easy way to specify a prefix for a prefix path in conjunction with `ServerExtension` at the moment - I think as-is is fine